connectors/sources/dropbox_aiohttp.py

```diff
@@ -0,0 +1,661 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Dropbox source module responsible to fetch documents from Dropbox online.
+"""
+import asyncio
+import json
+import os
+from datetime import datetime
+from enum import Enum
+from functools import cached_property, partial
+from urllib import parse
+
+import aiofiles
+import aiohttp
+from aiofiles.os import remove
+from aiofiles.tempfile import NamedTemporaryFile
+from aiohttp.client_exceptions import ClientResponseError, ServerDisconnectedError
+
+from connectors.logger import logger
+from connectors.source import BaseDataSource
+from connectors.utils import (
+    TIKA_SUPPORTED_FILETYPES,
+    CancellableSleeps,
+    RetryStrategy,
+    convert_to_b64,
+    evaluate_timedelta,
+    is_expired,
+    iso_utc,
+    retryable,
+)
+
+RETRY_COUNT = 3
+DEFAULT_RETRY_AFTER = 300  # seconds
+RETRY_INTERVAL = 2
+CHUNK_SIZE = 1024
+FILE_SIZE_LIMIT = 10485760  # ~10 Megabytes
+MAX_CONCURRENT_DOWNLOADS = 100
+LIMIT = 300  # Limit for fetching shared files per call
+PAPER = "paper"
+FILE = "File"
+FOLDER = "Folder"
+RECEIVED_FILE = "Received File"
+
+API_VERSION = 2
+
+if "RUNNING_FTEST" in os.environ:
+    logger.warning("x" * 100)
+    logger.warning(
+        f"DROPBOX CONNECTOR CALLS ARE REDIRECTED TO {os.environ['DROPBOX_API_URL']}"
+    )
+    logger.warning("IT'S SUPPOSED TO BE USED ONLY FOR TESTING")
+    logger.warning("x" * 100)
+    BASE_URLS = {
+        "ACCESS_TOKEN_BASE_URL": os.environ["DROPBOX_API_URL"],
+        "FILES_FOLDERS_BASE_URL": os.environ["DROPBOX_API_URL_V2"],
+        "DOWNLOAD_BASE_URL": os.environ["DROPBOX_API_URL_V2"],
+    }
+else:
+    BASE_URLS = {
+        "ACCESS_TOKEN_BASE_URL": "https://api.dropboxapi.com/",
+        "FILES_FOLDERS_BASE_URL": f"https://api.dropboxapi.com/{API_VERSION}/",
+        "DOWNLOAD_BASE_URL": f"https://content.dropboxapi.com/{API_VERSION}/",
+    }
+
+ENDPOINTS = {
+    "ACCESS_TOKEN": "oauth2/token",
+    "PING": "users/get_current_account",
+    "CHECK_PATH": "files/get_metadata",
+    "FILES_FOLDERS": "files/list_folder",
+    "FILES_FOLDERS_CONTINUE": "files/list_folder/continue",
+    "RECEIVED_FILES": "sharing/list_received_files",
+    "RECEIVED_FILES_CONTINUE": "sharing/list_received_files/continue",
+    "RECEIVED_FILE_METADATA": "sharing/get_shared_link_metadata",
+    "DOWNLOAD": "files/download",
+    "PAPER_FILE_DOWNLOAD": "files/export",
+    "RECEIVED_FILE_DOWNLOAD": "sharing/get_shared_link_file",
```

Comment:
should all these keys also be constants, so that the values can be accessed using constants, instead of with string literals? I can typo `"ACCESS_TOKEN"` just as easily as typoing `"oauth2/token"`, so  this currently isn't providing much value over just using the literal strings everywhere they're used.

Reply:
I think you misunderstood - I like having all the endpoints in a single place and referencing them easily. I am suggesting just adding some more constants, like:

```
ACCESS_TOKEN = "access_token"
...
ENDPOINTS = {
  ACCESS_TOKEN : "oauth2/token"
  ...
 }
```

So that they can be referenced below as `ENDPOINTS.get(ACCESS_TOKEN)` instead of `ENDPOINTS.get('ACCESS_TOKEN')`. This way, we get an error about the variable not existing if someone typos as `ACESS_TOKEN`, instead of just getting `None`.

Reply:
Gotcha! I'll create an enum for all the keys and refer them.